apps/ledger-live-mobile/src/logic/npsRatings.tsx

```diff
@@ -0,0 +1,346 @@
+import { useCallback } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { add, isBefore, parseISO } from "date-fns";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+import { accountsWithPositiveBalanceCountSelector } from "../reducers/accounts";
+import {
+  ratingsModalOpenSelector,
+  ratingsModalLockedSelector,
+  ratingsCurrentRouteNameSelector,
+  ratingsHappyMomentSelector,
+  ratingsDataOfUserSelector,
+} from "../reducers/ratings";
+import {
+  setRatingsModalOpen,
+  setRatingsCurrentRouteName,
+  setRatingsHappyMoment,
+  setRatingsDataOfUser,
+} from "../actions/ratings";
+import { track } from "../analytics";
+import { setNotificationsModalLocked } from "../actions/notifications";
+import { setUserNps } from "../actions/settings";
+
+export type RatingsHappyMoment = {
+  timeout?: number;
+  /** Name of the route that will trigger the rating flow */
+  // eslint-disable-next-line camelcase
+  route_name: string;
+  /** In milliseconds, delay before triggering the rating flow */
+  timer?: number;
+  /** Whether the rating flow is triggered when entering or when leaving the screen */
+  type?: "on_enter" | "on_leave";
+};
+
+export type RatingsDataOfUser = {
+  /** Date of the first time the user oppened the app */
+  appFirstStartDate?: Date;
+  /** Number of times the user oppened the application */
+  numberOfAppStarts?: number;
+  /** Number of times the user oppened the application since the last time his app crashed */
+  numberOfAppStartsSinceLastCrash?: number;
+  /** If set, we will not prompt the rating flow again before this date unless the user triggers it manually from the settings */
+  dateOfNextAllowedRequest?: Date;
+  /** Whether or not the user clicked on the "Not now" cta from the Enjoy step of the ratings flow */
+  alreadyClosedFromEnjoyStep?: boolean;
+  /** Whether or not the user clicked on the "Not now" cta from the Init step of the ratings flow */
+  alreadyClosedFromInitStep?: boolean;
+  /** Whether or not the user already rated the app */
+  alreadyRated?: boolean;
+  /** If true, we will not prompt the rating flow again unless the user triggers it manually from the settings */
+  doNotAskAgain?: boolean;
+  /** "satisfied" if user clicked on the "Satisfied" cta in the ratings flow, "disappointed" if user clicked on the "I'm disappointed" in the ratings flow */
+  satisfaction?: string;
+};
+
+const ratingsDataOfUserAsyncStorageKey = "ratingsDataOfUser";
```

Comment:
Should we update this storage key to be different than the one of the original ratings flow ?